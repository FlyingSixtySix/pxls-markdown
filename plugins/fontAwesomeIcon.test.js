const { processor, expectProcessorParse } = require('../testUtil.js');
const fontAwesome = require('./fontAwesomeIcon.js');

processor.use(fontAwesome);

test(
	'tokenizes "[fa]" as a FontAwesome icon',
	() => expectProcessorParse('[fa]').toMatchSnapshot()
);
test(
	'tokenizes "[fab.fa-discord]" correctly',
	() => expectProcessorParse('[fab.fa-discord]').toMatchSnapshot()
);
