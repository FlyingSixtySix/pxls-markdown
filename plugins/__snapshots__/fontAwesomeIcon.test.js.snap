// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tokenizes "[fa]" as a FontAwesome icon 1`] = `
Array [
  Object {
    "children": Array [
      Object {
        "classes": Array [
          "fa",
        ],
        "type": "fontAwesomeIcon",
      },
    ],
    "type": "paragraph",
  },
]
`;

exports[`tokenizes "[fab.fa-discord]" correctly 1`] = `
Array [
  Object {
    "children": Array [
      Object {
        "classes": Array [
          "fab",
          "fa-discord",
        ],
        "type": "fontAwesomeIcon",
      },
    ],
    "type": "paragraph",
  },
]
`;
